### PR TITLE
[EGD-7956] Fix DMA issue on tx modem data

### PR DESCRIPTION
**Pull request by lucckb**
Fix DMA issue on tx modem data.

### CODE REVIEW

**Review by Maciej-Mudita — APPROVED** [review]

**Review by wojtekrzepecki — APPROVED** [review]

File: module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp [hunk 1]
```diff
@@ -221,15 +221,20 @@ namespace bsp
...
         }
 
-        constexpr auto maxTXCheckRetires = 5;
+        constexpr auto maxTXCheckRetires = 10;
+        int retries;
         if (LPUART_IsEDMATxBusy(&uartDmaHandle)) {
-            for (int i = 0; i < maxTXCheckRetires; ++i) {
-                LOG_INFO("Waiting for TX free %d", i);
+            for (retries = 0; retries < maxTXCheckRetires; ++retries) {
+                LOG_INFO("Waiting for TX free %d", retries);
```
Comment (SP2FET) [inline module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp:228]: but... is this a real fix or just a workaround? It seems that giving it more retires doesn't fix the source of the issue...

File: module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp [hunk 2]
```diff
@@ -221,15 +221,20 @@ namespace bsp
             assert(false);
         }
 
-        constexpr auto maxTXCheckRetires = 5;
+        constexpr auto maxTXCheckRetires = 10;
```
Comment (SP2FET) [inline module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp:224]:
```suggestion
        constexpr auto maxTXCheckRetries = 10;
```

File: module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp (hunk 1 above)
Comment (lucckb) [reply module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp:228]: this pr is not related to retries count rather for exit on failure

**Review by niepiekm — APPROVED** [review]

File: module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp (hunk 1 above)
Comment (mjan-dev) [reply module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp]: @lucckb How is it related to exit on failure? It is not stated in issue and no other isssue is linked. Could you please elaborate.

**Review by mjan-dev — COMMENTED** [review]
Will the error handling be addressed in this PR or in another one concerning the same issue or another decision has been made? As stated in the description: "We should fix error handling if DMA is busy and still determine why it is busy for such a long time."

File: module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp [hunk 3]
```diff
@@ -221,15 +221,20 @@ namespace bsp
             assert(false);
         }
 
-        constexpr auto maxTXCheckRetires = 5;
+        constexpr auto maxTXCheckRetries = 10;
+        int retries;
```
Comment (alekrudnik) [inline module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp:225]:
```suggestion
        int retries = 0;
```

**Review by alekrudnik — APPROVED** [review]

File: module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp (hunk 3 above)
Comment (lucckb) [reply module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp:225]: it is set I the for look we don't need to set it to zero here

**Pull request merged**